routers/api/v1/repo/branch.go

```diff
@@ -396,6 +396,95 @@ func ListBranches(ctx *context.APIContext) {
 	ctx.JSON(http.StatusOK, apiBranches)
 }
 
+// RenameBranch renames a repository's branch.
+func RenameBranch(ctx *context.APIContext) {
+	// swagger:operation POST /repos/{owner}/{repo}/branches/{name}/rename repository repoRenameBranch
```

Comment:
The previous PR did have unresolved feedback [here](https://github.com/go-gitea/gitea/pull/23061#discussion_r1113878851).

If I understand correctly, the question is if having a branch "test/branch/rename" would cause our http routing to instead look for "test/branch"? If this was the question, this shouldn't be the case. The endpoint would like the following (but URL-encoded):
```
/api/v1/:owner/:repo/branches/test/branch/rename/rename
```

So this should be fine. From manually testing with our swagger client, this appears to be case.

Reply:
https://github.com/go-gitea/gitea/pull/32433/files#r1831575229